opm/autodiff/BlackoilWellModel_impl.hpp

```diff
@@ -1724,6 +1723,85 @@ namespace Opm {
                 }
                 ++local_comp_index;
             }
+
+            if (handle_ms_well && !well.segments.empty()) {
+                // we need the well_ecl_ information
+                const std::string& well_name = wm.first;
+                const Well* well_ecl = getWellEcl(well_name);
+                assert(well_ecl);
```

Comment:
`assert` here is kind of temporary approach. I have not been able to investigate the exception system introduced in https://github.com/OPM/opm-simulators/pull/1730. If it is needed to follow the exception system, I will fix the `assert` here and the one below.